cmd/buildah/umount.go

```diff
@@ -6,48 +6,90 @@ import (
 
 	"github.com/pkg/errors"
 	"github.com/projectatomic/buildah/util"
+	"github.com/sirupsen/logrus"
 	"github.com/urfave/cli"
 )
 
 var (
+	umountFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all, a",
+			Usage: "umount all of the currently mounted containers",
+		},
+	}
 	umountCommand = cli.Command{
 		Name:           "umount",
 		Aliases:        []string{"unmount"},
 		Usage:          "Unmounts the root file system on the specified working containers",
 		Description:    "Unmounts the root file system on the specified working containers",
 		Action:         umountCmd,
 		ArgsUsage:      "CONTAINER-NAME-OR-ID [...]",
+		Flags:          umountFlags,
 		SkipArgReorder: true,
 	}
 )
 
 func umountCmd(c *cli.Context) error {
+	umountAll := c.Bool("all")
 	umountContainerErrStr := "error unmounting container"
 	args := c.Args()
-	if len(args) == 0 {
+	if len(args) == 0 && !umountAll {
 		return errors.Errorf("at least one container ID must be specified")
 	}
+	if len(args) > 0 && umountAll {
+		return errors.Errorf("when using the --all switch, you may not pass any container IDs")
+	}
 
 	store, err := getStore(c)
 	if err != nil {
 		return err
 	}
 
 	var lastError error
-	for _, name := range args {
+	if len(args) > 0 {
+		for _, name := range args {
+			builder, err := openBuilder(getContext(), store, name)
+			if err != nil {
+				if lastError != nil {
+					logrus.Error(lastError)
+				}
+				lastError = util.WriteError(os.Stderr, errors.Wrapf(err, "%s %s", umountContainerErrStr, name), lastError)
```

Comment:
This looks wrong, I believe the util.WriteError will print the error to the screen so you will end up with lastErrror reported twice.  I thin the code should probbaly look like.

if lastError != nil {
		util.WriteError(os.Stderr, lastError)
}
lastError = errors.Wrapf(err, "%s %s", umountContainerErrStr, name)

Reply:
Done.